### PR TITLE
Indexing fixes

### PR DESCRIPTION
**Pull request by isaacsas**

### CODE REVIEW

**Comment by isaacsas** [comment]
@AayushSabharwal can you check I've got the correct dispatches now for mutating parameters within JumpProblems ensuring `update_parameters!(...)` is called to update the associated MassActionJump?

**Comment by isaacsas** [comment]
If this looks ok I'll add some more symbolic tests for integrators and for remake.

File: src/problem.jl [hunk 1]
```diff
@@ -122,9 +122,17 @@ end
...
+
+# for updating parameters in JumpProblems to update MassActionJumps
+function SII.set_parameter!(prob::JumpProblem, val, idx)
+    ans = SII.set_parameter!(SII.parameter_values(prob), val, idx)
+
     if using_params(prob.massaction_jump)
         update_parameters!(prob.massaction_jump, prob.prob.p)
     end
+
+    ans
```
Comment (AayushSabharwal) [inline src/problem.jl:135]: While this is technically correct, `set_parameter!` will be called once for each parameter which would make this _very_ expensive. I'll PR to SII to add a callback that runs at the end of `setp` (`finalize_parameters_hook!(prob, ps)`?)

Comment (AayushSabharwal) [reply src/problem.jl:135]: https://github.com/SciML/SymbolicIndexingInterface.jl/pull/58

Comment (isaacsas) [reply src/problem.jl:135]: @AayushSabharwal please ping me when a SII release with that callback is available so I can get this finished -- we need this for making updates to Catalyst for MTK9. Thanks!

Comment (AayushSabharwal) [reply src/problem.jl:135]: It's been released now @isaacsas

Comment (AayushSabharwal) [reply src/problem.jl:135]: Oops, I thought it was merged. Sorry 😅

Comment (AayushSabharwal) [reply src/problem.jl:135]: _Now_ it's out in 0.3.13

**Comment by isaacsas** [comment]
@ChrisRackauckas this hadn't yet been updated for https://github.com/SciML/SymbolicIndexingInterface.jl/pull/58, which is the interface we are supposed to use (and was only merged/released today).

I'll open another PR to update this to the correct interface, hopefully later today or tomorrow.

**Pull request merged**